From 178078cac0c1d649983b091d97280f6a66d04e62 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Wed, 26 Feb 2020 15:08:13 +0200
Subject: [PATCH] PONRTSYS-7025:
 tcp-upstream-and-downstream-traffic-is-not-flowing-properly-in-dslite-tunnel

Fix PPPoE length diff calculation in PPv4 session manager

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
index be519c3c490b..606cc889ba48 100644
--- a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
+++ b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
@@ -2898,7 +2898,7 @@ static void __smgr_pppoe_mod_set(struct sess_info *sess)
 		 */
 		if (__smgr_is_iptun_encp_sess(sess) &&
 		    __smgr_is_ipv6_pkt(SESS_EGRS_OUTER_BMAP(sess)))
-			sce->pppoe_diff += sess->args->nhdr->outer_len.l3_len;
+			sce->pppoe_diff += sizeof(struct ipv6hdr);
 
 		sce->l2_org_val = 0;
 		/* calculate the pppoe header offset */
